src/Bicep.Core/Registry/Providers/TypesV1Archive.cs

```diff
@@ -44,4 +49,21 @@ private static async Task AddFileToTar(TarWriter tarWriter, string archivePath,
 
         await tarWriter.WriteEntryAsync(tarEntry);
     }
+
+    private static HashSet<string> getAllUniqueTypePaths(string pathToIndex)
+    {
+        FileStream indexStream = new(pathToIndex, FileMode.Open, FileAccess.Read);
+
+        var index = TypeSerializer.DeserializeIndex(indexStream);
+
+        HashSet<string> uniqueTypePaths = new();
+
+        foreach (var typeInformation in index.Resources)
+        {
+            uniqueTypePaths.Add(Path.GetDirectoryName(pathToIndex) + "\\" + typeInformation.Value.RelativePath);
```

Comment:
I would avoid storing the parent path here - you'll want to use the 'relative path' directly when saving files into the archive
```suggestion
            uniqueTypePaths.Add(typeInformation.Value.RelativePath);
```